### PR TITLE
 Init Project: Setup .env, database, and logging

### LINKED ISSUE #8: Init Project: Setup .env, database, and logging
For this ticket we will need to do 3 things:

- [x] Create .env file called .env. This .env file will not be checked into git as information to the database should be kept private. Will be used for db migrations . The `DATABASE_URL` is used by dbmate to run migrations against the database/
    ```
    DATABASE_URL="postgres://postgres:[postgres password]@host.docker.internal:[host port]/[databasename]?sslmode=disable"
    DEBUG=true
    ```
- [x] Install and create a init migration using [dbmate](https://github.com/amacneil/dbmate).
    ```
     dbmate new init
    ```
- [x] Create a config.go file that will include all the logic for reading in environment variables following the [12 factor app](https://12factor.net/) approach. Use [iken/config](https://github.com/bir/iken) to read in environment variables based on a struct. 
    ```
    type Config struct {
	LocalDebug       bool   `env:"DEBUG"`
	Port             int    `env:"PORT, 8080"`
	Host             string `env:"HOST, 0.0.0.0"`
	DatabaseURL      string `env:"DATABASE_URL"`
     }

    func LoadConfig() *Config {
	c := Config{}
	if err := config.Load(&c); err != nil {
		log.Fatal().Err(err).Msg("error loading config")
	}

	if c.Port <= 0 || c.Port > 65535 {
		log.Fatal().Int("port", c.Port).Msg("invalid Port provided")
	}

	return &c
    }
    ```
- [x] Use [zerolog](https://github.com/rs/zerolog) for our logging. Note: Make sure all logs end with a `.Msg("")` or `.Send()`
    ```
    // If we are in debug, use the console writer to pretty print the logs
    log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
    zerolog.SetGlobalLevel(zerolog.DebugLevel)
    ```
    
See Parent ticket (https://github.com/SAKA-club/todo/issues/6 ) for more details about Init Project.

### PR DESCRIPTION
**Pull request by klooopmoo**
closes #8 
Server runs and connects using the config file rather than hard coding. .env file updated to include db_url. Verified dbmate connects to the database using the URL. Zerologging included debug set to true

### CODE REVIEW

**Comment by spencerreeves** [comment]
@klooopmoo make sure the title of the PR is descriptive of the PR itself. Uses `closes #8`, not covers. If you want to read more about the reserved words in Github, checkout this [document](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

File: backend/db/migrations/20220510225729_init.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- migrate:up
+
+
+-- migrate:down
+
```
Comment (spencerreeves) [inline backend/db/migrations/20220510225729_init.sql]: If the file is empty, omit it from the PR.

File: backend/main.go [hunk 2]
```diff
@@ -28,11 +42,24 @@ func main() {
...
+	var err error
+
+	host, err = os.Hostname()
+	if err != nil {
+		log.Panic().Err(err).Msg("unable to get hostname")
+	}
+
+	zerolog.SetGlobalLevel(zerolog.DebugLevel)
+
+	log.Log().Str("Host", host).Msg("Service Startup")
```
Comment (spencerreeves) [inline backend/main.go]: This should be right before we start our server.

File: backend/main.go [hunk 3]
```diff
@@ -28,11 +42,24 @@ func main() {
...
+
+func initLog(cfg *Config) {
+	var err error
+
+	host, err = os.Hostname()
+	if err != nil {
+		log.Panic().Err(err).Msg("unable to get hostname")
+	}
+
+	zerolog.SetGlobalLevel(zerolog.DebugLevel)
```
Comment (spencerreeves) [inline backend/main.go]: Let's move all of this to the main function. No need to have a separate function call for these changes.

Comment (spencerreeves) [inline backend/main.go]: If we are in debug mode, lets set the level to `Debug`, otherwise set it to `Info`.

File: backend/swagger.yaml [hunk 4]
```diff
@@ -20,3 +22,175 @@ paths:
...
+      title:
+        type: string
+      body:
+        type: string
+      priority:
+        type: boolean
+      finish_by:
+        type: string
+        format: date
+        example: "2022-05-05"
```
Comment (spencerreeves) [inline backend/swagger.yaml]: These changes seem to correlate better with #10 . Let's hold off on adding these in until we pickup that ticket.

File: backend/main.go (hunk 3 above)
Comment (klooopmoo) [reply backend/main.go]: im not sure what you mean by set it to info? is this a flag i should have in my .env file?

**Review by spencerreeves — APPROVED** [review]

**Pull request merged**